src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java

```diff
@@ -661,6 +661,10 @@ protected Object mapObject(JsonParser p, DeserializationContext ctxt,
     {
         private static final long serialVersionUID = 1L;
 
+        // Arbitrarily chosen.
+        // Introduced to resolve CVE-2020-36518 and as a temporary hotfix for #2816
+        private static final int MAX_DEPTH = 1000;
```

Comment:
@cowtowncoder @TaylorSMarks would it make sense to add a public static method to UntypedObjectDeserializer.java that allows users to set a different limit (but still with 1000 default)?